src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -1,23 +1,25 @@
 from __future__ import annotations
 
-from snowflake.cli.api.project.schemas import (
-    native_app,
-    snowpark,
-    streamlit,
-)
-from snowflake.cli.api.project.schemas.relaxed_map import RelaxedMap
-from strictyaml import (
-    Int,
-    Optional,
-)
+from typing import Optional
 
-project_schema = RelaxedMap(
-    {
-        "definition_version": Int(),
-        Optional("native_app"): native_app.native_app_schema,
-        Optional("snowpark"): snowpark.snowpark_schema,
-        Optional("streamlit"): streamlit.streamlit_schema,
-    }
-)
+from pydantic import Field
+from snowflake.cli.api.project.schemas.native_app import NativeAppSchema
+from snowflake.cli.api.project.schemas.snowpark import SnowparkSchema
+from snowflake.cli.api.project.schemas.streamlit import StreamlitSchema
+from snowflake.cli.api.project.schemas.updatable_model import UpdatableModel
 
-project_override_schema = project_schema.as_fully_optional()
+
+class ProjectDefinitionSchema(UpdatableModel):
+    definition_version: int = Field(
+        title="Version of the project definition schema, which is currently 1"
+    )
+    native_app: Optional[NativeAppSchema] = Field(
+        title="Native app definitions for the project", default=None
+    )
+    snowpark: Optional[SnowparkSchema] = Field(
+        title="Snowpark functions and procedures definitions for the project",
+        default=None,
+    )
+    streamlit: Optional[StreamlitSchema] = Field(
+        title="Native app definitions for the project", default=None
+    )
```

Comment:
Let's add validator for `definition_version`

Reply:
you mean `definition == 1`?

Reply:
Another way to think about this -- when we get to project definition v2 (design currently in progress) it will look completely different. We should have a way to separate out the definitions by `definition_version` rather than have to have a complex base model that has to support both. I agree that for now ensuring `definition_version == 1` is important, but we should make sure that when we introduce `definition_version == 2` it's something that can live alongside this model rather than "within" it.

Reply:
Ok, so my proposition is to add `definition_version==1` to field validation. If user provides another version of definition, it's pointless to proceed with parsing any fields.
Then, when we add new versions, we can change `load_project_definition()` logic to decide which model should be used for provided data.

Reply:
I agree, let's have `==1` for now. We will add more validation once we start working on v2.

Reply:
Added